File: Sorting Algorithms/bubbleSort.js
const bubbleSort =(array)=>{
    for(let i=0;i<array.length-1;i++){
        let flag=false
        for(let j=0;j<array.length-i-1;j++){
            if(array[j]>array[j+1]){
                flag=true
                let temp = array[j];
                array[j]=array[j+1];
                array[j+1]=temp;
            }
        }
        if(!flag) break
    }
    return array
}

const array=[12,34,56,45,87,67,-9,7]
console.log(bubbleSort(array));

File: src/sorting/BubbleSort.java
package sorting;

public class BubbleSort {
    public static void main(String[] args) {
        int[] array = {12, 343, 45, 23, 0, 9, 34, 6, 5, 7, 8};
        int[] sortedArray = bubbleSort(array);
        for (int i = 0; i < sortedArray.length; i++) {
            System.out.print(" " + sortedArray[i]);
        }
    }

    private static int[] bubbleSort(int[] array) {
        for (int i = 0; i < array.length; i++) {
            for (int j = 0; j < array.length; j++) {
                if (array[i] < array[j]) {
                    int temp = array[i];
                    array[i] = array[j];
                    array[j] = temp;
                }
            }
        }
        return array;
    }
}
